assets/blocks/lesson-actions/lesson-actions-block/edit.js

```diff
@@ -135,18 +140,29 @@ const EditLessonActionsBlock = ( {
 				blockName: 'sensei-lms/button-reset-lesson',
 				label: __( 'Reset lesson', 'sensei-lms' ),
 			},
+			{
+				blockName: 'sensei-lms/button-view-quiz',
+			},
 		],
 	} );
 
+	useHasQuiz( {
+		quizToggle: toggleBlocks.find(
+			( i ) => i.blockName === 'sensei-lms/button-view-quiz'
+		).onToggle,
+	} );
+
 	// Filter inner blocks based on the settings.
 	const filteredInnerBlocksTemplate = INNER_BLOCKS_TEMPLATE.filter(
 		( i ) => false !== toggledBlocks[ i[ 0 ] ]
 	);
 
+	const userToggleBlocks = toggleBlocks.filter( ( i ) => false !== i.label );
```

Comment:
I'm thinking a different idea for that. Instead checking the label, we could use the `useHasQuiz` hook to return a boolean before the `useToggleBlocks`, and we could add the block in the `blocks` array for the `useToggleBlocks`, only if it's true. WDYT?

Reply:
Handled in refactor.